### PR TITLE
Sprint 5: OS Initialization fix for Rhel 

### PR DESCRIPTION
**Pull request by surajsbharadwaj**
### Description

Please include a summary of the change and which issue is fixed. Also, please include the motivation and context for the change. List any dependencies that are required for this change.

**Check the relevant boxes:**
- [x] Bug fix (nonbreaking change that fixes an issue)
- [ ] New feature (nonbreaking change that adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Examples or tests (addition or updates of examples or tests)
- [x] Documentation update
- [ ] CI related update (pipeline, etc.)

### Checklist

- [ ] If relevant, a test for the change has been added or updated as part of this PR.
- [ ] If relevant, documentation for the change has been added or updated as part of this PR.

### Merge

- Merge using "Squash and merge".
- Make sure to use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents. The commit message determines whether a new version of the modules needs to be released, and if so, which semver number to use).

### CODE REVIEW

File: examples/basic/variables.tf [hunk 1]
```diff
@@ -65,13 +61,13 @@ variable "transit_gateway_name" {
...
   type        = bool
-  default     = true
+  default     = false
 }
 
 variable "cloud_connection_count" {
   description = "Required number of Cloud connections to create or reuse. The maximum number of connections is two per location."
   type        = number
-  default     = 0
+  default     = 2
```
Comment (ocofaigh) [inline examples/basic/variables.tf:70]: should we just use `1` for the basic example?

Comment (surajsbharadwaj) [reply examples/basic/variables.tf:70]: Its okay we can use 2. no harm :)

File: README.md [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
 - Attaches the private networks to the IBM Cloud connections
 
 Following limitations currently apply:
-- The only number of IBM Cloud connections we support is two.
+- The only number of IBM Cloud connections we support is **two**.
 - Reuse of IBM Cloud connections is not supported.
-- Private networks in IBM PowerVS service must be in 10.0.0.0/8 range
-- Only SLES15 SP3 as operating system is supported
+- Private networks in IBM PowerVS Workspace must be in 10.0.0.0/8 range
+- **Only SLES15 SP3 and RHEL 8.4 as operating system is supported**
```
Comment (Beilmann) [inline README.md:15]: Only SLES15 SP3 and RHEL 8.4 as operating system **are** supported

File: README.md [hunk 3]
```diff
@@ -65,7 +65,7 @@ module "power-infrastructure" {
 |------|---------|
 | <a name="requirement_terraform"></a> [terraform](#requirement\_terraform) | >= 1.1.0 |
 | <a name="requirement_ibm"></a> [ibm](#requirement\_ibm) | >= 1.44.3 |
-| <a name="requirement_time"></a> [time](#requirement\_time) | 0.8.0 |
+| <a name="requirement_time"></a> [time](#requirement\_time) | >= 0.8.0 |
```
Comment (Beilmann) [inline README.md:68]: Is that stable enough to rely on a newest (and probably not tested) dependency?

File: examples/ibm-catalog/standard-solution/README.md [hunk 4]
```diff
@@ -51,19 +51,19 @@ Make sure you installed the Power flavor of version 0.0.19 or higher of Secure L
...
+| <a name="input_prerequisite_workspace_id"></a> [prerequisite\_workspace\_id](#input\_prerequisite\_workspace\_id) | IBM Cloud Schematics workspace ID of an existing Secure infrastructure on VPC for regulated industries with VSIs deployment. If you do not yet have an existing deployment, click [here](https://cloud.ibm.com/catalog/content/slz-vpc-with-vsis-a87ed9a5-d130-47a3-980b-5ceb1d4f9280-global#create) to create one. When you deploy this module, use this [configuration](https://raw.githubusercontent.com/terraform-ibm-modules/terraform-ibm-powervs-infrastructure/main/examples/ibm-catalog/standard-solution/slz_json_configs_for_powervs/). Copy the configuration from the link into the `override_json_string` deployment value. | `string` | n/a | yes |
```
Comment (Beilmann) [inline examples/ibm-catalog/standard-solution/README.md:62]: use one of supported [configurations]

File: examples/ibm-catalog/standard-solution/variables.tf [hunk 5]
```diff
@@ -1,15 +1,11 @@
...
+  description = "IBM Cloud Schematics workspace ID of an existing Secure infrastructure on VPC for regulated industries with VSIs deployment. If you do not yet have an existing deployment, click [here](https://cloud.ibm.com/catalog/content/slz-vpc-with-vsis-a87ed9a5-d130-47a3-980b-5ceb1d4f9280-global#create) to create one. When you deploy this module, use this [configuration](https://raw.githubusercontent.com/terraform-ibm-modules/terraform-ibm-powervs-infrastructure/main/examples/ibm-catalog/standard-solution/slz_json_configs_for_powervs/). Copy the configuration from the link into the `override_json_string` deployment value."
```
Comment (Beilmann) [inline examples/ibm-catalog/standard-solution/variables.tf:2]: one of supported [configurations]

File: version.tf [hunk 6]
```diff
@@ -13,7 +13,7 @@ terraform {
     }
     time = {
       source  = "hashicorp/time"
-      version = "0.8.0"
+      version = ">= 0.8.0"
```
Comment (Beilmann) [inline version.tf:16]: is that safe to rely on a not tested version?

File: README.md (hunk 3 above)
Comment (surajsbharadwaj) [reply README.md:68]: @ocofaigh  suggested to add a range here.
Any comments ?

File: README.md (hunk 2 above)
Comment (surajsbharadwaj) [reply README.md:15]: Updated

File: examples/ibm-catalog/standard-solution/README.md (hunk 4 above)
Comment (surajsbharadwaj) [reply examples/ibm-catalog/standard-solution/README.md:62]: updated

File: examples/ibm-catalog/standard-solution/variables.tf (hunk 5 above)
Comment (surajsbharadwaj) [reply examples/ibm-catalog/standard-solution/variables.tf:2]: done

**Comment by surajsbharadwaj** [comment]
@ocofaigh @Beilmann please review and merge. 

Version:
- Added full support for RHEL
- Renamed variables
- Sample variable file added:
- Readmes added for submodule

**Comment by surajsbharadwaj** [comment]
@ocofaigh NOtes:

- Add support for RHEL8.4 (VPC and PVS parts)
- Timeout during RHEL registration process while PVS Infra deployment's module power_management_service_ntp
- awscli installation failed for RHEL
- RHEL overwrite json added
- Review default variables value for standard examples
- Output proxy_host_or_ip_port from terraform-ibm-powervs-infrastructure should match squid_host_or_ip in terraform-ibm-powervs-sap
- restrict zone values in validation for deployment

**Review by Beilmann — APPROVED** [review]

**Pull request merged**